app/src/main/java/com/android/wildex/model/cache/usersettings/UserSettingsDataStore.kt

```diff
@@ -0,0 +1,54 @@
+package com.android.wildex.model.cache.usersettings
+
+import android.content.Context
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import com.android.wildex.model.user.AppearanceMode
+import kotlinx.coroutines.flow.first
+
+val Context.userSettingsDataStore by preferencesDataStore(name = "user_settings_cache")
+
+/** Preference keys for user settings. */
+object UserSettingsPreferencesKey {
+  val ENABLE_NOTIFICATIONS = booleanPreferencesKey("enable_notifications")
+  val APPEARANCE_MODE = intPreferencesKey("appearance_mode")
+}
+
+/** Implementation of IUserSettingsCache using DataStore for persistent storage. */
+class UserSettingsCache(private val context: Context) : IUserSettingsCache {
+  override suspend fun initializeUserSettings() {
```

Comment:
And refactor accordingly


```suggestion
  override suspend fun initializeUserSettings(userId: String) {
    setEnableNotification(userId, true)
    setAppearanceMode(userId, AppearanceMode.AUTOMATIC)
  }

  override suspend fun getEnableNotification(userId: String): Boolean? {
    val prefs = context.userSettingsDataStore.data.first()
    return prefs[enableNotificationsKey(userId)]
  }

  override suspend fun setEnableNotification(userId: String, enable: Boolean) {
    context.userSettingsDataStore.edit { prefs -> prefs[enableNotificationsKey(userId)] = enable }
  }

  override suspend fun getAppearanceMode(userId: String): AppearanceMode? {
    val prefs = context.userSettingsDataStore.data.first()
    return prefs[appearanceModeKey(userId)]?.let { AppearanceMode.entries[it] }
  }

  override suspend fun setAppearanceMode(userId: String, mode: AppearanceMode) {
    context.userSettingsDataStore.edit { prefs -> prefs[appearanceModeKey(userId)] = mode.ordinal }
  }

  override suspend fun clear(userId: String) {
    context.userSettingsDataStore.edit { prefs ->
      prefs.remove(enableNotificationsKey(userId))
      prefs.remove(appearanceModeKey(userId))
    }
  }
```